src/examples/avg.d

```diff
@@ -0,0 +1,68 @@
+/* This file contains an example on how to use the tranzitive visitor.
+   It implements a visitor which computes the average function length from
+   a *.d file.
+ */
+
+module examples.avg;
+
+import ddmd.astbase;
+import ddmd.parse;
+import ddmd.transitivevisitor;
+
+import ddmd.globals;
+import ddmd.id;
+import ddmd.identifier;
+
+import std.stdio;
+import std.file;
+
+class FunctionLengthVisitor : TransitiveVisitor
+{
+    alias visit = super.visit;
+    ulong[] lengths;
+
+    float getAvgLen(ASTBase.Module m)
+    {
+        m.accept(this);
+
+        if (lengths.length == 0)
+            return 0;
+
+        ulong sum = 0;
+        foreach (len; lengths)
+            sum += len;
+
+        //writeln(lengths);
+        return sum/lengths.length;
+    }
+
+    override void visitFuncBody(ASTBase.FuncDeclaration fd)
+    {
+        lengths ~= fd.endloc.linnum - fd.loc.linnum;
+        super.visitFuncBody(fd);
+    }
+}
+
+void main()
+{
+
+    string fname = "examples/testavg.d";
+
+    Id.initialize();
+    global._init();
+    global.params.isLinux = true;
+    global.params.is64bit = (size_t.sizeof == 8);
+    global.params.useUnitTests = true;
+    ASTBase.Type._init();
```

Comment:
That's quite a lot of initialization needed. We should discuss how to reduce that.